crates/api/src/entrypoint.rs

```diff
@@ -1,30 +1,157 @@
 use crate::{deploy::DeployFile, manifest::WasmImage, wasmtime_unit::Wasmtime};
 
+use std::env;
 use std::path::{Component, Path, PathBuf};
 
 use anyhow::{bail, Result};
 use log::info;
 
-/// Validates the deployed image.
-///
-/// Takes path to the workdir as an argument.
-pub fn start(workdir: impl AsRef<Path>) -> Result<()> {
-    let workdir = workdir.as_ref();
-    let deploy_file = DeployFile::load(workdir)?;
+const INIT_MEM_VAR: &str = "YA_RUNTIME_WASI_INIT_MEM";
+const OPTIMIZE_VAR: &str = "YA_RUNTIME_WASI_OPT";
+const SGX_VAR: &str = "YA_RUNTIME_WASI_SGX";
 
-    info!(
-        "Validating deployed image {:?}.",
-        get_log_path(workdir, deploy_file.image_path())
-    );
+/// [Yagna] WASI runtime configuration.
+#[derive(Default, Clone, Debug)]
+pub struct RuntimeOptions {
+    pub(crate) max_static_memory: Option<u64>,
+    pub(crate) optimize: Option<bool>,
+    pub(crate) sgx_profile: Option<bool>,
+}
 
-    let mut image = WasmImage::new(&deploy_file.image_path())?;
-    let mut wasmtime = create_wasmtime(workdir, &mut image, &deploy_file)?;
+impl RuntimeOptions {
+    /** Initializes runtime options from environment variables.
+     *
+     * * `YA_RUNTIME_WASI_INIT_MEM` - maximum memory size. (supported formats 250m, 1.2g)
+     * * `YA_RUNTIME_WASI_OPT` - optimization. (0|no for no optimalization), (1|yes)
+     * * `YA_RUNTIME_WASI_SGX` - enables sgx profiled configuration.
+     */
+    pub fn from_env() -> Result<Self> {
+        let mut me = Self::default();
 
-    wasmtime.load_binaries(&mut image)?;
+        if let Some(err_msg) = (|| {
+            let mem_str = env::var(INIT_MEM_VAR).ok()?;
+            let len = mem_str.as_bytes().len();
+            let scale = match mem_str.as_bytes().get(len - 1) {
+                Some(b'k') => 0x400,
+                Some(b'm') => 0x100_000,
+                Some(b'g') => 0x40_000_000,
+                _ => return Some(format!("invalid max mem spec: {}", mem_str)),
+            };
+            let value = match mem_str[..len - 1].parse::<u64>() {
+                Ok(val) => val,
+                Err(e) => return Some(format!("invalid max mem spec: {} ({})", mem_str, e)),
+            };
+            me.max_static_memory = Some(value * scale);
+            None
+        })() {
+            log::warn!("wasi env MAX_MEM_VAR {}", err_msg);
```

Comment:
Given that we already pull in `log::info` into scope, it might make more sense to do the same with `log::warn`.

Reply:
`use` combined with warn(unused-imports) makes me angry. evry time i change single message from `info` to `warn` if i add use. then if i change to error or back to info or remove this line it will generate warning.  so every tuning i logs requires to check use in begining of file. i have this. for me `log::info!` & `log::warn!` it is a clean and convenient notation. and I've been using it since rust supports it. it's not long like `std::path::PathBuf`. `use log::warn;` & `warn!(..)` are much longer than just `log::warn!(...)`.

Reply:
Makes sense!